rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/model/LRARecord.java

```diff
@@ -419,17 +432,21 @@ private boolean afterLRARequest(URI target, String payload) {
                 builder.header(LRA.LRA_HTTP_CONTEXT_HEADER, lra.getId().toASCIIString());
             }
 
-            builder.async(PARTICIPANT_TIMEOUT, TimeUnit.SECONDS);
-
-            ResponseHolder response = target.equals(forgetURI) ? builder.delete() : builder.put(payload, MediaType.TEXT_PLAIN);
+            Future<Response> responseFuture =  target.equals(forgetURI) ? builder.async().delete()
+                : builder.async().put(Entity.text(payload));
+            Response response = responseFuture.get(PARTICIPANT_TIMEOUT, TimeUnit.SECONDS);
 
             if (response.getStatus() == 200) {
                 return true;
             }
-        } catch (WebApplicationException e) {
+        } catch (Exception e) {
```

Comment:
WebApplicationException is the exception of choice here - since we are using JAX-RS.

Reply:
The other exceptions come from the Future.get call.

Reply:
Yes I know but the WebApplicationException will include the jax-rs error even for the Future.get call.

Reply:
not sure what you are proposing. The `e.getMessage()` will contain all the info from WebApplicationException in case. Or do you propose to throw WebApplicationException here?

Reply:
My point is that the original code was fine as it was.

Reply:
I will leave this thread unresolved since I still argue that the original code was fine.